js/src/logger.ts

```diff
@@ -779,8 +780,27 @@ type BackgroundLogEvent =
 export interface DatasetRecord {
   id: string;
   input: any;
-  output: any;
+  expected?: any;
   metadata: any;
+  output?: any;
```

Comment:
Maybe helpful to be super strict about types here and define it as
```
export type DatasetRecord =
    & { id, input, metadata }
    & ({ expected?: any } | { output?: any })
```

Reply:
Explicitly typed both versions instead